packages/material-ui/src/styles/createTypography.d.ts

```diff
@@ -18,23 +18,23 @@ export type ThemeStyle =
 
 export interface FontStyle
   extends Required<{
-    fontFamily: CSSProperties['fontFamily'];
+    fontFamily: React.CSSProperties['fontFamily'];
     fontSize: number;
-    fontWeightLight: CSSProperties['fontWeight'];
-    fontWeightRegular: CSSProperties['fontWeight'];
-    fontWeightMedium: CSSProperties['fontWeight'];
-    fontWeightBold: CSSProperties['fontWeight'];
+    fontWeightLight: React.CSSProperties['fontWeight'];
+    fontWeightRegular: React.CSSProperties['fontWeight'];
+    fontWeightMedium: React.CSSProperties['fontWeight'];
+    fontWeightBold: React.CSSProperties['fontWeight'];
   }> {}
 
 export interface FontStyleOptions extends Partial<FontStyle> {
   htmlFontSize?: number;
-  allVariants?: CSSProperties;
+  allVariants?: React.CSSProperties;
```

Comment:
The object is applied with CSS-in-JS solutions, for instance, developers could use media queries with. In this context, is it OK?

Reply:
Just to be clear: We don't document this at all. Considering that some styles in the theme are applied via CSS-in-JS and some not I would rather reduce the type to the vanilla React.CSSProperties. We'll see if this is actually used.

Reply:
We partially document it in https://material-ui.com/customization/typography/#responsive-font-sizes

Reply:
> It depends on what you mean by used

What the word means. 

Seems like #14573 did not add a proper documentation or types for this behavior. Otherwise I would expect an error there.

Reply:
Definitely, there is a grey area. To be clear, my previous comment had the individual typography variants in mind. I doubt `allVariants` is frequently used

More importantly, as we prepare the migration to support styled-components and emotion out of the box. How can we update the types here to support these style engines?

Reply:
If there is no error, does it mean we are missing a coverage with TypeScript for the support of it?

Reply:
Probably. I'll add a test and see how we can reconcile.

Reply:
With #19269 it seems like this PR does not impact #14573. Do you have another case that I might have missed in #19269?

Reply:
This seems safe to apply now since it looks like we don't document nor test this behavior.